src/modules/flow/http-server/http-server.c

```diff
@@ -0,0 +1,626 @@
+
+    r = sol_http_param_add(&response.param, SOL_HTTP_REQUEST_PARAM_HEADER(
+        HTTP_HEADER_CONTENT_TYPE, (send_json) ? HTTP_HEADER_CONTENT_TYPE_JSON : HTTP_HEADER_CONTENT_TYPE_TEXT));
+    SOL_INT_CHECK_GOTO(r, != true, end);
+
+    r = sol_buffer_set_slice(&response.content, slice);
+    SOL_INT_CHECK_GOTO(r, < 0, end);
+
+    if (method == SOL_HTTP_METHOD_POST) {
+        sol_flow_send_irange_packet(mdata->node, SOL_FLOW_NODE_TYPE_HTTP_SERVER_INT__OUT__OUT, &mdata->value.i);
+    }
+
+end:
+    sol_buffer_fini(&response.content);
+    sol_http_param_free(&response.param);
+
+    return r;
+}
+
+static int
+int_process(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    struct http_data *mdata = data;
+    int r;
+
+    r = sol_flow_packet_get_irange(packet, &mdata->value.i);
+    SOL_INT_CHECK(r, < 0, r);
+
+    r = sol_http_server_set_last_modified(server, mdata->path, time(NULL));
+    SOL_INT_CHECK(r, < 0, r);
+
+    return 0;
+}
+
+static void
+int_close(struct sol_flow_node *node, void *data)
+{
+    struct http_data *mdata = data;
+
+    stop_server(mdata);
+}
+
+static int
+int_open(struct sol_flow_node *node, void *data, const struct sol_flow_node_options *options)
+{
+    int r;
+    struct http_data *mdata = data;
+    struct sol_flow_node_type_http_server_int_options *opts =
+        (struct sol_flow_node_type_http_server_int_options *)options;
+
+    mdata->value.i = opts->value;
+    mdata->node = node;
+
+    r = start_server(mdata, opts->path, int_response_cb);
+    SOL_INT_CHECK(r, < 0, r);
+
+    return 0;
+}
+
+/* ------------------------------------------- float ------------------------------------------------- */
+
+static int
+float_response_cb(void *data, struct sol_http_request *request)
+{
+    int r;
+    uint16_t idx;
+    enum sol_http_method method;
+    bool send_json = false, modified = false;
+    struct http_data *mdata = data;
+    char str[512] = { 0 };
+    struct sol_str_slice slice = { .data = str };
+    struct sol_http_param_value *value;
+    struct sol_http_response response = {
+        .api_version = SOL_HTTP_RESPONSE_API_VERSION,
+        .content = SOL_BUFFER_INIT_EMPTY,
+        .param = SOL_HTTP_REQUEST_PARAM_INIT,
+        .response_code = SOL_HTTP_STATUS_OK
+    };
+
+    method = sol_http_request_get_method(request);
+    response.url = sol_http_request_get_url(request);
+
+#define STRTOD_(field_) \
+    do { \
+        errno = 0; \
+        modified = true; \
+        mdata->value.d.field_ = strtod(value->value.key_value.value, NULL); \
+        if (errno != 0) \
+            return -errno; \
+    } while (0)
+
+    SOL_HTTP_PARAM_FOREACH_IDX (sol_http_request_get_params(request), value, idx) {
+        switch (value->type) {
+        case SOL_HTTP_PARAM_POST_FIELD:
+            if (streq(value->value.key_value.key, "value"))
+                STRTOD_(val);
+
+            if (streq(value->value.key_value.key, "min"))
```

Comment:
nope, i'm considering that the request can contains any field of the irange/drange